src/TargetResourceHandler.cpp

```diff
@@ -13,78 +13,82 @@ TargetResourceHandler::TargetResourceHandler(const FileSystemPolicy& fileSystemP
 /**
  * @brief Determine target resource based on the request path and handle different file types and statuses accordingly.
  *
- * Finds a matching location block.
- * If no match is found set Status to StatusNotFound.
- * Set the targetResource as location.root + request.uri.path and check file type.
- * RegularFile: break, nothing more to do.
- * Directory: if no trailing slash, add it and set Status to StatusMovedPermanently.
- *            if index file is set, add it to the path and set internalRedirect to true.
- *            if autoindex is set, set autoindex to true.
- *            else set Status to StatusForbidden.
- * FileNotExist: set Status to StatusNotFound.
- * FileOther: set Status to StatusForbidden.
- * If internalRedirect is true, repeat the process.
- * If stat fails, set Status to StatusInternalServerError.
+ * Constructs a new LocatingInfo object with the provided Connection.
+ * Recursively searches for the target resource with locateTargetResource() using the information in the LocatingInfo.
+ * Uses updateConnection() to write found information into Connection.
  * @param connection Connection object.
- * @param request HTTP request.
  */
-void TargetResourceHandler::execute(Connection& connection, HTTPRequest& request)
+void TargetResourceHandler::execute(Connection& connection)
+{
+	LocatingInfo locInfo(connection);
```

Comment:
Right now the function locateTargetResource() receives and returns LocatingInfo by value: If we don't find an index file, we want to try again - with the information (path, activeLocation, targetResource) _before_ the search was executed. This works by saving the value return in a tmp variable, check it and if an index was found (or error occured) we overwrite the actual struct. If not we can just pass the original again for the next index.

Connection is a struct with many members, some of which are expensive to copy. Mainly `connection.m_buffer` and `connection.m_request.headers`. So copying Connection on potentially several layers of recursion could explode memory usage.

Alternatively we could pass Connection as a reference. We would then need to save some stuff before searching (recursively) the index file. If we don't find it set the values back and try the next one. 

Instead of having to temp save several values, I just repackage it and can then pass by value, since a copy of LocatingInfo is cheap.

Reply:
Fair enough, basically it is a collection of temps on its own.